lib/etcd_lib.py

```diff
@@ -47,6 +47,29 @@ def get_ingress_address(endpoint_name):
     return sorted(all_addrs, key=lambda i: ip_address(i).version)[0]
 
 
+def get_snapshot_count(snapshot_count, channel):
+    """Returns the snapshot count value
+
+    * check if the value is auto,
+        iff channel >=3.2 it will set 100'000 otherwhise it will set 10'000
+    * any other integer value will be set as it is
+
+    @param snapshot_count the value to set, could be a number or auto
+    @param channel the channel used by the charm
+    """
+    if snapshot_count == "auto":
+        if channel == "auto" or float(channel.split("/")[0]) >= 3.2:
+            return 100000
+        return 10000
+    snapshot_count_number = 0
+    try:
+        snapshot_count_number = int(snapshot_count)
+    except ValueError:
+        raise TypeError(f"{snapshot_count} value is not an integer number")
+
+    return snapshot_count_number
```

Comment:
Maybe we can change this to:
```suggestion
    try:
        return int(snapshot_count)
    except ValueError:
        raise TypeError(f"{snapshot_count} value is not an integer number")
```